### PR TITLE
[IT-1186] Add shared Nextflow development account under ScienceOU

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
I'm opening this PR to create a development AWS account for running Nextflow workflows. The intent is that the development account will used to determine the minimal policies, etc. required for the production account. This assumes that we will have more access to the development account than the production account. This account will also be used to test deployment from the `develop` branch in the [provisioner repository](https://github.com/Sage-Bionetworks/aws-nextflow-workflows). 

See #130 for the PR creating the corresponding development account. 

----

PR Checklist:
- [x] Describe and explain your intentions for this change
- [x] Setup pre-commit and run the validators (info in README.md)
  To validate files run: `pre-commit run --all-files`

### CODE REVIEW

**Review by zaro0508 — APPROVED** [review]

**Comment by zaro0508** [comment]
Please fix up your PR comment

File: org-formation/organization.yaml [hunk 1]
```diff
@@ -27,6 +27,7 @@ Organization:
         - !Ref SandboxAccount
         - !Ref MobileHealthDataEngineeringAccount
         - !Ref SageITAccount
+        - !Ref NextflowWorkflowsDevAccount
```
Comment (ahayden) [inline org-formation/organization.yaml:30]: Please check the comment on #130 about the naming convention.

**Review by ahayden — APPROVED** [review]
Seems ok, though these branches should be named after their tickets.

**Comment by BrunoGrandePhD** [comment]
@ahayden: I wasn't aware of the branch naming convention. Will do going forward!

**Pull request merged**